snapcraft/tests/plugins/test_catkin.py

```diff
@@ -629,6 +633,59 @@ def test_run_environment_with_underlay(self, run_mock):
         self.assertThat(environment, Contains('. {}'.format(
             os.path.join(plugin.rosdir, 'snapcraft-setup.sh'))))
 
+    def _evaluate_environment(self, predefinition=''):
+        plugin = catkin.CatkinPlugin('test-part', self.properties,
+                                     self.project_options)
+
+        python_path = os.path.join(
+            plugin.installdir, 'usr', 'lib', 'python2.7', 'dist-packages')
+        os.makedirs(python_path)
+
+        # Save plugin environment off into a file and read the evaluated
+        # version back in, thus obtaining the real environment
+        with tempfile.NamedTemporaryFile(mode='w+') as f:
+            f.write(predefinition)
+            f.write('\n'.join(['export ' + e for e in plugin.env(
+                plugin.installdir)]))
+            f.write('python3 -c "import os; print(dict(os.environ))"')
+            f.flush()
+            return ast.literal_eval(subprocess.check_output(
+                ['/bin/sh', f.name]).decode(
+                    sys.getfilesystemencoding()).strip())
+
+    def _assert_pythonpath_no_empty_segments(self, environment):
+        # Verify that the PYTHONPATH contains no empty segments
+        self.assertThat(environment, Contains('PYTHONPATH'))
+        segments = environment['PYTHONPATH'].split(':')
+        empty_segments = [s for s in segments if not s.strip()]
+        self.assertThat(
+            empty_segments, HasLength(0),
+            'PYTHONPATH unexpectedly contains empty segments: {}'.format(
+                environment['PYTHONPATH']))
+
+        return segments
+
+    def test_pythonpath_if_not_defined(self):
+        environment = self._evaluate_environment()
+
+        self._assert_pythonpath_no_empty_segments(environment)
+
+    def test_pythonpath_if_null(self):
+        environment = self._evaluate_environment(textwrap.dedent("""
+            export PYTHONPATH=
+        """))
+
+        self._assert_pythonpath_no_empty_segments(environment)
+
+    def test_pythonpath_if_not_empty(self):
+        environment = self._evaluate_environment(textwrap.dedent("""
+            export PYTHONPATH=foo
+        """))
+
+        # Verify that the PYTHONPATH contains no empty segments
+        segments = self._assert_pythonpath_no_empty_segments(environment)
```

Comment:
reading this is not straightforward, ideally you have a helper that returns segments.

Reply:
Yeah that was a little odd. Fixed.